### PR TITLE
AT-6121 Allow only PDF files for Task Order upload

### PR DESCRIPTION
**Pull request by jGonzalez-CCPO**
System requests and accepts only PDF files for Task Order upload.

Ticket: AT-6121

### CODE REVIEW

File: atat/forms/validators.py [hunk 1]
```diff
@@ -98,5 +98,13 @@ def _file_length(_form, field):
     return _file_length
 
 
+def alpha_numeric_name_pdf(message=translate("forms.task_order.filename_error"),):
+    return Regexp(
+        regex=r"^([A-Za-z0-9\-_ \.\/\\]*(\.pdf))?$",
```
Comment (kylelaker-ccpo) [inline atat/forms/validators.py:103]: I think there are a few improvements that can be made to this regex. As is, the empty string (`""`) would be accepted. So would `.pdf` without anything before the extension. I'd recommend more like:

```
^[A-Za-z0-9\-_ \.\/\\]+\.pdf$
```

This enforces that prior to the `.` we have at least one of the allowed characters. The empty string is not accepted. There aren't any capture groups to add complexity either.

See https://regex101.com/r/D4UfKa/1 and especially the "Unit Tests" on the sidebar on the left.

**Review by kylelaker-ccpo — CHANGES_REQUESTED** [review]
Nice addition! Small request to get the wording more in-sync between the two files and a question on the regex

File: atat/forms/validators.py (hunk 1 above)
Comment (kylelaker-ccpo) [reply atat/forms/validators.py:103]: Saw the Jira comment. If we need to accept the empty string, we can do:
```
^([A-Za-z0-9\-_ \.\/\\]+\.pdf)?$
```

Comment (jGonzalez-CCPO) [reply atat/forms/validators.py:103]: Read my comment on the ticket, due to various tests that depends on this to be an empty string i was forced to modify it to accept that alternative.

**Review by jasonburkert-ccpo — APPROVED** [review]
Peer reviewed.  Tested in local development.  Looks good.
Upload dialog behaves as expected on supported browsers, and files with file extensions of mixed case are accepted and can be uploaded.

image

File: js/lib/input_validations.js [hunk 2]
```diff
@@ -139,9 +139,9 @@ export default {
   },
   restrictedFileName: {
     mask: false,
-    match: /^[A-Za-z0-9\-_ \.]+$/,
+    match: /^[A-Za-z0-9\-_ \.\/\\]*\.pdf$/i,
     unmask: [],
     validationError:
-      'File names can only contain the characters A-Z, 0-9, space, hyphen, underscore, and period.',
+      'File names can only contain the characters A-Z, 0-9, space, hyphen, underscore, and period. And end in .pdf',
```
[the remark this answers is not in the document]
Comment (jasonburkert-ccpo) [reply js/lib/input_validations.js]: Confirmed this was addressed and suggested change adopted.

File: translations.yaml [hunk 3]
```diff
@@ -316,7 +316,7 @@ forms:
   task_order:
     upload_error: There was an error uploading your file. Please try again. If you encounter repeated problems uploading this file, please contact CCPO.
     size_error: "The file you have selected is too large. Please choose a file no larger than {file_size_limit}MB."
-    filename_error: File names can only contain the characters A-Z, 0-9, space, hyphen, underscore, and period.
+    filename_error: File names must end on 'pdf' and can only contain the characters A-Z, 0-9, space, hyphen, underscore, and period.
```
[the remark this answers is not in the document]
Comment (jasonburkert-ccpo) [reply translations.yaml]: Confirmed this was addressed and suggested change adopted.

**Review by jasonburkert-ccpo — APPROVED** [review]
Good.  This change will require a base file name.  Simply `.pdf` will no longer be accepted.

image

**Review by kylelaker-ccpo — APPROVED** [review]
Requests have been addressed. Thanks!

**Comment by jGonzalez-CCPO** [comment]
@kylelaker-ccpo @jasonburkert-ccpo i just re-test, format and rebase all the requested changes :)

**Comment by jasonburkert-ccpo** [comment]
Tested locally again. Looks good.

**Pull request merged**